server/src/main/java/org/elasticsearch/common/settings/SecureSetting.java

```diff
@@ -151,12 +161,34 @@ public static Setting<InputStream> secureFile(String name, Setting<InputStream>
         return new SecureFileSetting(name, fallback, properties);
     }
 
+    private static class FallbackStringSetting extends Setting<SecureString> {
```

Comment:
I think it deserves a bit of documentation + indication that it will be removed once we have secure settings implemented outside the keystore